doc/source/ray-core/gpu-objects.rst

```diff
@@ -0,0 +1,212 @@
+
+For ease of following along, we'll start with the `Gloo <https://github.com/pytorch/gloo>`__ transport, which can be used without any physical GPUs.
+
+
+Installation
+------------
+
+.. note::
+    Under construction.
+
+.. _gpu-objects-gloo:
+Usage with gloo (CPUs only)
+---------------------------
+
+Installation
+************
+
+.. note::
+    Under construction.
+
+Walkthrough
+***********
+
+To get started, define an actor class and a task that returns a ``torch.Tensor``:
+
+.. literalinclude:: doc_code/gpu_objects_gloo.py
+   :language: python
+   :start-after: __normal_example_start__
+   :end-before: __normal_example_end__
+
+As written, when the ``torch.Tensor`` is returned, it will be copied into Ray's CPU-based object store.
+For CPU-based tensors, this can require an expensive step to copy and serialize the object, while GPU-based tensors additionally require a copy to and from CPU memory.
+
+To enable GPU objects, use the ``tensor_transport`` option in the :func:`@ray.method <ray.method>` decorator.
+
+.. literalinclude:: doc_code/gpu_objects_gloo.py
+   :language: python
+   :start-after: __gloo_example_start__
+   :end-before: __gloo_example_end__
+
+This decorator can be added to any actor tasks that return a ``torch.Tensor``, or that return ``torch.Tensors`` nested inside other Python objects.
+Adding this decorator will change Ray's behavior in the following ways:
+
+1. When returning the tensor, Ray will store a *reference* to the tensor instead of copying it to CPU memory. If multiple tensors are found, then Ray will store a reference for each tensor.
+2. When the :class:`ray.ObjectRef` is passed to another task, Ray will use Gloo to transfer the tensor to the destination task.
+
+For (2) to work, we must first create a *collective group* of actors.
+
+Creating a collective group
+***************************
+
+.. Add API pages for collective group APIs.
+
+To create a collective group for use with GPU objects, we must:
+
+1. Create multiple Ray actors.
+2. Create a collective group on the actors using the :func:`ray.experimental.collective.create_collective_group <ray.experimental.collective.create_collective_group>` function. The tensor transport specified should match the one used in the :func:`@ray.method <ray.method>` decorator.
+
+Here is an example:
+
+.. literalinclude:: doc_code/gpu_objects_gloo.py
+   :language: python
+   :start-after: __gloo_group_start__
+   :end-before: __gloo_group_end__
+
+The actors can now communicate directly via gloo.
+The group can also be destroyed using the :func:`ray.experimental.collective.destroy_collective_group <ray.experimental.collective.destroy_collective_group>` function.
+After calling this function, the actors can be reused as normal and new collective groups can be created on them.
```

Comment:
This makes it seem like actors can't be used for anything else when they're part of a collective group